exercises/change/example.go

```diff
@@ -0,0 +1,191 @@
+	}
+	if target < 0 {
+		// No change possible for negative.
+		return nil, ErrNegativeTarget
+	}
+	findSolution(coins, target, &solution)
+	if len(solution) == 0 {
+		// No solution possible with coins given.
+		return nil, ErrNoSolutionFound
+	}
+	return solution, nil
+}
+
+// findSolution attempts to find a change solution for target amount
+// from the coins given.
+func findSolution(coins []int, target int, s *[]int) {
+	// Find the coins to consider, less than the target change amount.
+	last := len(coins) - 1
+	for ; last >= 0; last-- {
+		if coins[last] == target {
+			// Perfect single coin solution, so done!
+			*s = []int{target}
+			return
+		}
+		if coins[last] > target {
+			// Cannot use this too large coin.
+			continue
+		}
+		// Now, 0 .. last are the coins to consider.
+		break
+	}
+	if last < 0 {
+		// All coins larger than target.
+		return
+	}
+	findBestMixedSolution(coins[0:last+1], target, s)
+}
+
+// findBestMixedSolution attempts to find a best mixed coin amount solution
+// making correct change for the amount target.
+func findBestMixedSolution(coins []int, target int, s *[]int) {
+	// Consider all combinations of the coins.
+	// Eliminate the combination if it sums to value greater than target.
+	n := len(coins)
+	bestsize := target + 1 // an initial "worst" bestcase
+comboLoop:
+	for m := uint64((1 << uint(n)) - 1); m >= 1; m-- {
+		// Make a list, c, of the coins matching combination m,
+		// as long as the sum of coins in c doesn't exceed target.
+		sum := 0
+		c := make([]int, 0, n)
+		for b := 0; b < n; b++ {
+			// smaller b is larger coin
+			if (m & (1 << uint(b))) != 0 {
+				sum += coins[n-1-b]
+				if sum > target {
+					// Already sum exceeds target for this combination,
+					// so consider it no more.
+					continue comboLoop
+				}
+				c = append(c, coins[n-1-b])
+			}
+		}
+		// Now the combination of coins in c can be considered.
+		// The sum of coins in c doesn't exceed target.
+		// When the sum matches the target, those coins are a solution.
+		if sum == target {
+			updateSolution(c, s, &bestsize)
+			continue
+		}
+		// Recursively search for a best solution.
+		partialSolution(nil, c, target, s, &bestsize)
+	}
+}
+
+// updateSolution updates solution with newsolution if no prior solution
+// or when length of newsolution is better.
+func updateSolution(newsolution []int, solution *[]int, bestsize *int) {
+	if len(*solution) == 0 || len(newsolution) < len(*solution) {
+		*bestsize = len(newsolution)
+		*solution = make([]int, len(newsolution))
+		copy(*solution, newsolution)
+		sort.Ints(*solution)
+	}
+}
+
+// partialSolution takes a partial solution, and coin list c looking to meet target
+// while updating solution s to include partial plus the coins meeting target.
+func partialSolution(partial []int, c []int, target int, s *[]int, bestsize *int) {
+	// We want to consider coins in c in largest to smallest order.
+	sort.Sort(sort.Reverse(sort.IntSlice(c)))
+
+	for e := range c {
```

Comment:
optional: I notice `c[e]` being used a lot in here. Perhaps a name like `coin` could be given to it. Same number of characters.